src/Components/Navbar/Navbar.js

```diff
@@ -0,0 +1,113 @@
+import "./navbar.css";
+import SubscriptionsIcon from "@mui/icons-material/Subscriptions";
+import WatchLaterIcon from "@mui/icons-material/WatchLater";
+import HistoryIcon from "@mui/icons-material/History";
+import ThumbUpIcon from "@mui/icons-material/ThumbUp";
+import { Link } from "react-router-dom";
+import { useProducts } from "./../../Context/products-context";
+import { useEffect } from "react";
+import { useNavbar } from "./../../Context/navbar-context";
+import { useNavigate } from "react-router-dom";
+import { useToken } from "./../../Context/token-context";
+
+const Navbar = () => {
+  let navigate = useNavigate();
+
+  const { encodedToken, setEncodedToken } = useToken();
+
+  const { navbarButtonText, setNavbarButtonText } = useNavbar();
+
+  const { state, dispatch } = useProducts();
+
+  useEffect(() => {
+    localStorage.setItem("LIKED_ARRAY", JSON.stringify(state.likedArray));
+    localStorage.setItem(
+      "WATCH_LATER_ARRAY",
+      JSON.stringify(state.watchLaterArray)
+    );
+    localStorage.setItem("HISTORY_ARRAY", JSON.stringify(state.historyArray));
+    localStorage.setItem("DISLIKED_ARRAY", JSON.stringify(state.dislikedArray));
```

Comment:
you should do all of this stuff in the context or reducer of the respective data, here it will just run again and again